convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -79,13 +80,36 @@ def __call__(self, cmd, print_cmd=True, print_output=True):
             self.called += 1
             return self.output, self.ret_code
 
+    @unit_tests.mock(utils, "run_subprocess", RunSubprocessMocked())
+    @unit_tests.mock(tool_opts, "disablerepo", [])
+    @unit_tests.mock(tool_opts, "disable_submgr", True)
+    def test_call_yum_cmd_without_disablerepo_and_disable_submgr_true(self):
+        pkghandler.call_yum_cmd("install")
+
+        self.assertEqual(utils.run_subprocess.cmd, "yum install -y --disablerepo=*")
+
+    @unit_tests.mock(utils, "run_subprocess", RunSubprocessMocked())
+    @unit_tests.mock(tool_opts, "disablerepo", ['rhel-7-extras-rpm'])
+    @unit_tests.mock(tool_opts, "disable_submgr", False)
+    def test_call_yum_cmd_with_disablerepo_and_disable_submgr_false(self):
+        pkghandler.call_yum_cmd("install")
+
+        self.assertEqual(utils.run_subprocess.cmd, "yum install -y --disablerepo=rhel-7-extras-rpm")
+
+    @unit_tests.mock(utils, "run_subprocess", RunSubprocessMocked())
+    @unit_tests.mock(tool_opts, "disablerepo", ['rhel-7-extras-rpm'])
+    def test_call_yum_cmd_with_disablerepo(self):
+        pkghandler.call_yum_cmd("install")
+
+        self.assertEqual(utils.run_subprocess.cmd, "yum install -y --disablerepo=rhel-7-extras-rpm")
```

Comment:
Where in these three tests do you test the proper addition of `submgr_enabled_repos` as described by @SpyTec:

> Ah, I see. So that one covers `elif tool_opts.disablerepo:` but it doesn't cover 
> ```py
> if not tool_opts.disable_submgr:
>     for repo in system_info.submgr_enabled_repos:
>         repos.append(repo)
> ```
> For that two tests should be created
> 1. if `disable_submgr` is true, `system_info.submgr_enabled_repos` should not be appended to repos var.
> 1. if `disable_submgr` is false, `system_info.submgr_enabled_repos` should be appended to repos var.
> 
> Then add a different test for when the user has `--disablerepo` for a RHSM repo e.g. `--disablerepo=rhel-7-server-rpms`, we should not add this repo to `--enablerepo`. Otherwise we'll end up with a command that looks like `yum install -y --enablerepo=rhel-7-server-rpms --disablerepo=rhel-7-server-rpms` which is rather ambiguous and should be avoided.

Reply:
To add to my comment, there is missing functionality in the code right now where it should not add to `--disablerepo` if it's already added as `--enablerepo`. The tests I mentioned in my comment will catch this, i.e. will fail currently. You could utilize [TDD](https://en.wikipedia.org/wiki/Test-driven_development) approach and write the tests first and see that it is complete once the tests pass

Reply:
@fellipeh how's the progress on this?

Reply:
@fellipeh this is still not resolved